MySQL 5.5 Debian Stretch keys are expired

### Preliminary checklist

- [x] I am using the latest stable version of DDEV (see [upgrade guide](https://ddev.readthedocs.io/en/stable/users/install/ddev-upgrade/))
- [x] I have searched [existing issues](https://github.com/ddev/ddev/issues)
- [x] I have checked the [troubleshooting guide](https://ddev.readthedocs.io/en/stable/users/usage/troubleshooting/)
- [x] I have run `ddev debug test` to include output below

### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```
[COPY-PASTE HERE THE OUTPUT OF `ddev debug test`]
```
</details>


### Expected Behavior

It should work.

Related:

- #3004

### Actual Behavior

Stretch keys are expired:

```
$ docker run -it --rm --entrypoint=bash ddev/ddev-dbserver-mysql-5.5:v1.24.6 apt-key list
/etc/apt/trusted.gpg
--------------------
pub   rsa4096 2016-06-30 [SC]
      4D1B B29D 63D9 8E42 2B21  13B1 9334 A25F 8507 EFA5
uid           [ unknown] Percona Development Team (Packaging key) <info@percona.com>
sub   rsa4096 2016-06-30 [E]

/etc/apt/trusted.gpg.d/debian-archive-stretch-automatic.gpg
-----------------------------------------------------------
pub   rsa4096 2017-05-22 [SC] [expired: 2025-05-20]
      E1CF 20DD FFE4 B89E 8026  58F1 E0B1 1894 F66A EC98
uid           [ expired] Debian Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org>

/etc/apt/trusted.gpg.d/debian-archive-stretch-security-automatic.gpg
--------------------------------------------------------------------
pub   rsa4096 2017-05-22 [SC] [expired: 2025-05-20]
      6ED6 F5CB 5FA6 FB2F 460A  E88E EDA0 D238 8AE2 2BA9
uid           [ expired] Debian Security Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org>

/etc/apt/trusted.gpg.d/debian-archive-stretch-stable.gpg
--------------------------------------------------------
pub   rsa4096 2017-05-20 [SC] [expired: 2025-05-18]
      067E 3C45 6BAE 240A CEE8  8F6F EF0F 382A 1A7B 6500
uid           [ expired] Debian Stable Release Key (9/stretch) <debian-release@lists.debian.org>
```

### Steps To Reproduce

Container tests are failing https://github.com/ddev/ddev/actions/runs/15150754874

```
...
#13 1.493 W: GPG error: http://archive.debian.org/debian stretch Release: The following signatures were invalid: EXPKEYSIG 04EE7237B7D453EC Debian Archive Automatic Signing Key (9/stretch) <ftpmaster@debian.org> EXPKEYSIG EF0F382A1A7B6500 Debian Stable Release Key (9/stretch) <debian-release@lists.debian.org> The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 648ACFD622F3D138 NO_PUBKEY 0E98404D386FA1D9
#13 1.493 W: The repository 'http://archive.debian.org/debian stretch Release' is not signed.
...
 > [ 7/26] RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y -o Dpkg::Options::="--force-confold" bzip2 curl gnupg2 less lsb-release pv tzdata vim-tiny wget:
2.355 After this operation, 51.2 MB of additional disk space will be used.
2.355 WARNING: The following packages cannot be authenticated!
2.355   libssl1.1 libpython3.5-minimal libexpat1 python3.5-minimal python3-minimal
2.355   mime-support libmpdec2 libpython3.5-stdlib python3.5 libpython3-stdlib
2.355   dh-python python3 tzdata libssl1.0.2 xxd vim-common vim-tiny libunistring0
2.355   libidn2-0 libpsl5 wget bzip2 libmagic-mgc libmagic1 file krb5-locales less
2.355   libkeyutils1 libkrb5support0 libk5crypto3 libkrb5-3 libgssapi-krb5-2
2.355   xz-utils openssl ca-certificates libnghttp2-14 librtmp1 libssh2-1 libcurl3
2.355   curl distro-info-data gnupg2 lsb-release pv publicsuffix
2.356 E: There were unauthenticated packages and -y was used without --allow-unauthenticated
------
Dockerfile:46
--------------------
  44 |     RUN rm -f /etc/apt/sources.list.d/mariadb.list /etc/apt/sources.list.d/percona.list
  45 |     RUN mkdir -p /var/lib/apt/lists/partial
  46 | >>> RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y -o Dpkg::Options::="--force-confold" bzip2 curl gnupg2 less lsb-release pv tzdata vim-tiny wget
  47 |     RUN update-alternatives --install /usr/bin/vim vim /usr/bin/vim.tiny 10
  48 |     
--------------------
ERROR: failed to solve: process "/bin/bash -c apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y -o Dpkg::Options::=\"--force-confold\" bzip2 curl gnupg2 less lsb-release pv tzdata vim-tiny wget" did not complete successfully: exit code: 100
make: *** [Makefile:52: mysql_5.5_amd64] Error 1
```

### Anything else?

`check_key_expirations.sh` didn't catch it because it was disabled for MySQL 5.5 and 5.6:

https://github.com/ddev/ddev/blob/2842925c37731d0b623a8148fccb8952c44c6909/containers/ddev-dbserver/test/image_general.bats#L15-L26

--- a/containers/ddev-dbserver/Dockerfile
+++ b/containers/ddev-dbserver/Dockerfile
@@ -23,13 +23,16 @@ SHELL ["/bin/bash", "-c"]
 
 # Fix APT for Debian Stretch (EOL; upstream mirrors disabled)
 # Based on: https://serverfault.com/a/1131653
-RUN if grep "Debian GNU/Linux 9" /etc/issue >/dev/null; then \
-    rm -f /etc/apt/sources.list.d/mysql.list && \
-    echo "deb http://archive.debian.org/debian/ stretch main contrib non-free" >/etc/apt/sources.list && \
-    echo "deb http://archive.debian.org/debian-security/ stretch/updates main contrib non-free" >>/etc/apt/sources.list && \
-    apt-get -qq update -o Acquire::AllowInsecureRepositories=true -o Acquire::AllowDowngradeToInsecureRepositories=true -o APT::Get::AllowUnauthenticated=true && \
-    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends --no-install-suggests -o APT::Get::AllowUnauthenticated=true debian-archive-keyring; \
+RUN <<EOF
+    set -eu -o pipefail
+    if grep "Debian GNU/Linux 9" /etc/issue >/dev/null; then
+        rm -f /etc/apt/sources.list.d/mysql.list
+        echo "deb http://archive.debian.org/debian/ stretch main contrib non-free" >/etc/apt/sources.list
+        echo "deb http://archive.debian.org/debian-security/ stretch/updates main contrib non-free" >>/etc/apt/sources.list
+        apt-get -qq update -o Acquire::AllowInsecureRepositories=true -o Acquire::AllowDowngradeToInsecureRepositories=true -o APT::Get::AllowUnauthenticated=true
+        DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends --no-install-suggests -o APT::Get::AllowUnauthenticated=true debian-archive-keyring
     fi
+EOF
 
 # MariaDB 11.x moved MySQL symlinks into separate packages
 RUN set -x; if ( command -v mariadbd && ! command -v mysqld ); then \

--- a/containers/ddev-dbserver/Dockerfile
+++ b/containers/ddev-dbserver/Dockerfile
@@ -21,14 +21,15 @@ ARG MYSQL_ROOT_PASSWORD=root
 
 SHELL ["/bin/bash", "-c"]
 
-# Debian Stretch archives have been turned off
-RUN if grep "Debian GNU/Linux 9" /etc/issue >/dev/null ; then \
-    echo "deb http://archive.debian.org/debian/ stretch main contrib non-free" >/etc/apt/sources.list; \
+# Fix APT for Debian Stretch (EOL; upstream mirrors disabled)
+# Based on: https://serverfault.com/a/1131653
+RUN if grep "Debian GNU/Linux 9" /etc/issue >/dev/null; then \
+    rm -f /etc/apt/sources.list.d/mysql.list && \
+    echo "deb http://archive.debian.org/debian/ stretch main contrib non-free" >/etc/apt/sources.list && \
+    echo "deb http://archive.debian.org/debian-security/ stretch/updates main contrib non-free" >>/etc/apt/sources.list && \
+    apt-get -qq update -o Acquire::AllowInsecureRepositories=true -o Acquire::AllowDowngradeToInsecureRepositories=true -o APT::Get::AllowUnauthenticated=true && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends --no-install-suggests -o APT::Get::AllowUnauthenticated=true debian-archive-keyring; \
     fi
-# Remove obsolete MySQL 5.5/5.6 Jessie and before keys so they don't make expiration key test stumble
-RUN for item in "75DD C3C4 A499 F1A1 8CB5  F3C8 CBF8 D6FD 518E 17E1" "126C 0D24 BD8A 2942 CC7D  F8AC 7638 D044 2B90 D010" "D211 6914 1CEC D440 F2EB  8DDA 9D6D 8F6B C857 C906" "A1BD 8E9D 78F7 FE5C 3E65  D8AF 8B48 AD62 4692 5553" "ED6D 6527 1AAC F0FF 15D1  2303 6FB2 A1C2 65FF B764"; do \
-    apt-key remove "${item}" || true; \
-  done;
 
 # MariaDB 11.x moved MySQL symlinks into separate packages
 RUN set -x; if ( command -v mariadbd && ! command -v mysqld ); then \

--- a/pkg/ddevapp/config.go
+++ b/pkg/ddevapp/config.go
@@ -1184,21 +1184,49 @@ stopasgroup=true
 	extraDBContent := ""
 	if app.Database.Type == nodeps.Postgres {
 		extraDBContent = extraDBContent + fmt.Sprintf(`
-RUN set -e; source /etc/os-release; if [ "${VERSION_CODENAME:-}" = "stretch" ] || [ "${VERSION_CODENAME:-}" = "buster" ]; then \
-	rm -f /etc/apt/sources.list.d/pgdg.list; \
-    echo "deb http://archive.debian.org/debian/ ${VERSION_CODENAME} main contrib non-free" >/etc/apt/sources.list; \
-	[ "${VERSION_CODENAME}" = "stretch" ] && echo "deb http://archive.debian.org/debian-security/ ${VERSION_CODENAME}/updates main contrib non-free" >>/etc/apt/sources.list; \
-    (timeout %s apt-get -qq update -o Acquire::AllowInsecureRepositories=true -o Acquire::AllowDowngradeToInsecureRepositories=true -o APT::Get::AllowUnauthenticated=true || true); \
-    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends --no-install-suggests -o APT::Get::AllowUnauthenticated=true debian-archive-keyring apt-transport-https ca-certificates; \
-	echo "deb http://apt-archive.postgresql.org/pub/repos/apt/ ${VERSION_CODENAME}-pgdg-archive main" >/etc/apt/sources.list.d/pgdg.list; \
-    fi
 ENV PATH=$PATH:/usr/lib/postgresql/$PG_MAJOR/bin
 ADD postgres_healthcheck.sh /
-RUN chmod ugo+rx /postgres_healthcheck.sh
-RUN mkdir -p /etc/postgresql/conf.d && chmod 777 /etc/postgresql/conf.d
-RUN echo "*:*:db:db:db" > ~postgres/.pgpass && chown postgres:postgres ~postgres/.pgpass && chmod 600 ~postgres/.pgpass && chmod 777 /var/tmp && ln -sf /mnt/ddev_config/postgres/postgresql.conf /etc/postgresql && echo "restore_command = 'true'" >> /var/lib/postgresql/recovery.conf
-RUN printf "# TYPE DATABASE USER CIDR-ADDRESS  METHOD \nhost  all  all 0.0.0.0/0 md5\nlocal all all trust\nhost    replication    db             0.0.0.0/0  trust\nhost replication all 0.0.0.0/0 trust\nlocal replication all trust\nlocal replication all peer\n" >/etc/postgresql/pg_hba.conf
-RUN (timeout %s apt-get update || true) && DEBIAN_FRONTEND=noninteractive apt-get install -y -o Dpkg::Options::="--force-confold" --no-install-recommends --no-install-suggests bzip2 less procps pv vim
+RUN <<EOF
+set -eu -o pipefail
+source /etc/os-release
+
+if [ "${VERSION_CODENAME:-}" = "stretch" ] || [ "${VERSION_CODENAME:-}" = "buster" ]; then
+    rm -f /etc/apt/sources.list.d/pgdg.list
+    echo "deb http://archive.debian.org/debian/ ${VERSION_CODENAME} main contrib non-free" >/etc/apt/sources.list
+    if [ "${VERSION_CODENAME:-}" = "stretch" ]; then
+        echo "deb http://archive.debian.org/debian-security/ ${VERSION_CODENAME}/updates main contrib non-free" >>/etc/apt/sources.list
+    fi
+    timeout %s apt-get -qq update -o Acquire::AllowInsecureRepositories=true \
+        -o Acquire::AllowDowngradeToInsecureRepositories=true -o APT::Get::AllowUnauthenticated=true || true
+    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends --no-install-suggests -o APT::Get::AllowUnauthenticated=true \
+        debian-archive-keyring apt-transport-https ca-certificates
+    echo "deb http://apt-archive.postgresql.org/pub/repos/apt/ ${VERSION_CODENAME}-pgdg-archive main" >/etc/apt/sources.list.d/pgdg.list
+fi
+
+chmod ugo+rx /postgres_healthcheck.sh
+mkdir -p /etc/postgresql/conf.d
+chmod 777 /etc/postgresql/conf.d
+echo "*:*:db:db:db" > ~postgres/.pgpass
+chown postgres:postgres ~postgres/.pgpass
+chmod 600 ~postgres/.pgpass
+chmod 777 /var/tmp
+ln -sf /mnt/ddev_config/postgres/postgresql.conf /etc/postgresql
+echo "restore_command = 'true'" >>/var/lib/postgresql/recovery.conf
+
+echo "# TYPE DATABASE USER CIDR-ADDRESS  METHOD
+host  all         all 0.0.0.0/0 md5
+local all         all trust
+host  replication db  0.0.0.0/0 trust
+host  replication all 0.0.0.0/0 trust
+local replication all trust
+local replication all peer" >/etc/postgresql/pg_hba.conf
+
+timeout %s apt-get update || true
+DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    -o Dpkg::Options::="--force-confold" --no-install-recommends --no-install-suggests \
+    apt-transport-https bzip2 ca-certificates less procps pv vim-tiny
+update-alternatives --install /usr/bin/vim vim /usr/bin/vim.tiny 10
+EOF
 `, app.GetMinimalContainerTimeout(), app.GetMinimalContainerTimeout())
 	}
 

--- a/pkg/ddevapp/config.go
+++ b/pkg/ddevapp/config.go
@@ -1177,24 +1177,21 @@ stopasgroup=true
 		return "", err
 	}
 
-	// Add .pgpass to homedir on PostgreSQL
+	// Include PostgreSQL support for EOL Debian releases
+	// debian-security updates added for "stretch" only because this archive is not available for "buster" yet
+	// Based on: https://serverfault.com/a/1131653
+	// And add .pgpass to homedir on PostgreSQL
 	extraDBContent := ""
 	if app.Database.Type == nodeps.Postgres {
-		// PostgreSQL 9/10/11 upstream images are stretch-based, out of support from Debian.
-		// PostgreSQL 9/10 are out of support by PostgreSQL and no new images being pushed, see
-		// https://github.com/docker-library/postgres/issues/1012
-		// However, they do have a postgres:11-bullseye, but we won't start using it yet
-		// because of awkward changes to $DBIMAGE. PostgreSQL 11 will be EOL Nov 2023
-		if nodeps.ArrayContainsString([]string{nodeps.Postgres9, nodeps.Postgres10, nodeps.Postgres11}, app.Database.Version) {
-			extraDBContent = extraDBContent + fmt.Sprintf(`
-RUN rm -f /etc/apt/sources.list.d/pgdg.list
-RUN echo "deb http://archive.debian.org/debian/ stretch main contrib non-free" > /etc/apt/sources.list
-RUN timeout %s apt-get update || true
-RUN apt-get -y install apt-transport-https
-RUN printf "deb http://apt-archive.postgresql.org/pub/repos/apt/ stretch-pgdg main" > /etc/apt/sources.list.d/pgdg.list
-`, app.GetMinimalContainerTimeout())
-		}
 		extraDBContent = extraDBContent + fmt.Sprintf(`
+RUN set -e; source /etc/os-release; if [ "${VERSION_CODENAME:-}" = "stretch" ] || [ "${VERSION_CODENAME:-}" = "buster" ]; then \
+	rm -f /etc/apt/sources.list.d/pgdg.list; \
+    echo "deb http://archive.debian.org/debian/ ${VERSION_CODENAME} main contrib non-free" >/etc/apt/sources.list; \
+	[ "${VERSION_CODENAME}" = "stretch" ] && echo "deb http://archive.debian.org/debian-security/ ${VERSION_CODENAME}/updates main contrib non-free" >>/etc/apt/sources.list; \
+    (timeout %s apt-get -qq update -o Acquire::AllowInsecureRepositories=true -o Acquire::AllowDowngradeToInsecureRepositories=true -o APT::Get::AllowUnauthenticated=true || true); \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends --no-install-suggests -o APT::Get::AllowUnauthenticated=true debian-archive-keyring apt-transport-https ca-certificates; \
+	echo "deb http://apt-archive.postgresql.org/pub/repos/apt/ ${VERSION_CODENAME}-pgdg-archive main" >/etc/apt/sources.list.d/pgdg.list; \
+    fi
 ENV PATH=$PATH:/usr/lib/postgresql/$PG_MAJOR/bin
 ADD postgres_healthcheck.sh /
 RUN chmod ugo+rx /postgres_healthcheck.sh
@@ -1202,7 +1199,7 @@ RUN mkdir -p /etc/postgresql/conf.d && chmod 777 /etc/postgresql/conf.d
 RUN echo "*:*:db:db:db" > ~postgres/.pgpass && chown postgres:postgres ~postgres/.pgpass && chmod 600 ~postgres/.pgpass && chmod 777 /var/tmp && ln -sf /mnt/ddev_config/postgres/postgresql.conf /etc/postgresql && echo "restore_command = 'true'" >> /var/lib/postgresql/recovery.conf
 RUN printf "# TYPE DATABASE USER CIDR-ADDRESS  METHOD \nhost  all  all 0.0.0.0/0 md5\nlocal all all trust\nhost    replication    db             0.0.0.0/0  trust\nhost replication all 0.0.0.0/0 trust\nlocal replication all trust\nlocal replication all peer\n" >/etc/postgresql/pg_hba.conf
 RUN (timeout %s apt-get update || true) && DEBIAN_FRONTEND=noninteractive apt-get install -y -o Dpkg::Options::="--force-confold" --no-install-recommends --no-install-suggests bzip2 less procps pv vim
-`, app.GetMinimalContainerTimeout())
+`, app.GetMinimalContainerTimeout(), app.GetMinimalContainerTimeout())
 	}
 
 	err = WriteBuildDockerfile(app, app.GetConfigPath(".dbimageBuild/Dockerfile"), app.GetConfigPath("db-build"), app.DBImageExtraPackages, "", extraDBContent)

--- a/pkg/ddevapp/config.go
+++ b/pkg/ddevapp/config.go
@@ -1186,10 +1186,10 @@ stopasgroup=true
 		extraDBContent = extraDBContent + fmt.Sprintf(`
 ENV PATH=$PATH:/usr/lib/postgresql/$PG_MAJOR/bin
 ADD postgres_healthcheck.sh /
+
 RUN <<EOF
 set -eu -o pipefail
-source /etc/os-release
-
+source /etc/os-release || true
 if [ "${VERSION_CODENAME:-}" = "stretch" ] || [ "${VERSION_CODENAME:-}" = "buster" ]; then
     rm -f /etc/apt/sources.list.d/pgdg.list
     echo "deb http://archive.debian.org/debian/ ${VERSION_CODENAME} main contrib non-free" >/etc/apt/sources.list
@@ -1202,7 +1202,10 @@ if [ "${VERSION_CODENAME:-}" = "stretch" ] || [ "${VERSION_CODENAME:-}" = "buste
         debian-archive-keyring apt-transport-https ca-certificates
     echo "deb http://apt-archive.postgresql.org/pub/repos/apt/ ${VERSION_CODENAME}-pgdg-archive main" >/etc/apt/sources.list.d/pgdg.list
 fi
+EOF
 
+RUN <<EOF
+set -eu -o pipefail
 chmod ugo+rx /postgres_healthcheck.sh
 mkdir -p /etc/postgresql/conf.d
 chmod 777 /etc/postgresql/conf.d

--- a/pkg/versionconstants/versionconstants.go
+++ b/pkg/versionconstants/versionconstants.go
@@ -17,7 +17,7 @@ var WebTag = "v1.24.6" // Note that this can be overridden by make
 var DBImg = "ddev/ddev-dbserver"
 
 // BaseDBTag is the main tag, DBTag is constructed from it
-var BaseDBTag = "v1.24.6"
+var BaseDBTag = "20250521_stasadev_archived_debian"
 
 // TraefikRouterImage is image for router
 var TraefikRouterImage = "ddev/ddev-traefik-router"

